### ISSUE TITLE
query(Terraform): false positive - kms key policy

### ISSUE THREAD

**Issue by rdkls** [Checkmarx/kics#6707]
### Platform
Terraform

### Provider
AWS

### Description
Query 
7ebc9038-0bde-479a-acc4-6ed7b6758899
https://docs.kics.io/latest/queries/terraform-queries/aws/7ebc9038-0bde-479a-acc4-6ed7b6758899

Generates false positive when using Terraform's new way to attach KMS Key policies using aws_kms_key_policy resource instead of inline

**Comment by kicsbot** [comment]
Please, follow the guideline for an issue title: 

For **bug**:

`bug(<scope>): <title starting with lowercase letter>`

For **query**:

`query(<platform>): <title starting with lowercase letter>`

For **feature request**:

`feat(<scope>): <title starting with lowercase letter>`

Thank you!
*KICS Team*

**Comment by freitasmillena** [comment]
Hi, @rdkls!

Thank you so much for contributing and helping to make KICS better :)
Can you please provide us the example file that generates false positive? 
Also, if you want, feel free to collaborate to the issue by opening a PR and we will review as soon as possible!

Thank you, 
KICS Team.

**Comment by biker2o** [comment]
@freitasmillena See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kms_key_policy

`resource "aws_kms_key" "example" {
  description = "example"
}

resource "aws_kms_key_policy" "example" {
  key_id = aws_kms_key.example.id
  policy = jsonencode({
    Id = "example"
    Statement = [
      {
        Action = "kms:*"
        Effect = "Allow"
        Principal = {
          AWS = "*"
        }

        Resource = "*"
        Sid      = "Enable IAM User Permissions"
      },
    ]
    Version = "2012-10-17"
  })
}`

**Comment by buehmann** [comment]
We are running into this as well. Here is an example of a false positive:
```tf
resource "aws_kms_key" "change" {
  description             = "KMS CMK key used for SNS encryption topic - change request for support"
  deletion_window_in_days = 30
  key_usage               = "ENCRYPT_DECRYPT"
  enable_key_rotation     = true
}

resource "aws_kms_key_policy" "change" {
  key_id = aws_kms_key.change.key_id
  policy = data.aws_iam_policy_document.sns_change_key.json
}

data "aws_iam_policy_document" "sns_change_key" {
  statement {
    # ...
  }

  statement {
    # ...
  }
}
```
```
bin/kics scan -p kms-key-policy.tf -o . --include-queries 7ebc9038-0bde-479a-acc4-6ed7b6758899
```
[results.json](https://github.com/user-attachments/files/28052481/results.json)
